File: builder/c-sharp/LessonDirector.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Builder
{
    // Builder arayüzünü kullanarak nesne örneklemesi yapar.
    // UML diyagramındaki Director'a denk gelmektedir.
    class LessonDirector
    {
        private LessonBuilder lessonBuilder;

        public LessonDirector(LessonBuilder lessonBuilder)
        {
            this.lessonBuilder = lessonBuilder;
        }

        public void Make()
        {
            lessonBuilder.GetLesson();
            lessonBuilder.ApplyDiscount();
            lessonBuilder.AddLessonNote();
        }
    }

}

File: builder/go/lesson-director.go
package main

// Builder arayüzünü kullanarak nesne örneklemesi yapar.
// UML diyagramındaki Director'a denk gelmektedir.
type lessonDirector struct {
	lessonBuilder LessonBuilder
}

func NewLessonDirector(lessonBuilder LessonBuilder) *lessonDirector {
	return &lessonDirector{lessonBuilder: lessonBuilder}
}

func (l *lessonDirector) make() {
	l.lessonBuilder.getLesson()
	l.lessonBuilder.applyDiscount()
	l.lessonBuilder.addLessonNote()
}
